pkg/master/master.go

```diff
@@ -581,7 +581,8 @@ func (m *Master) getServersToValidate(c *Config) map[string]apiserver.Server {
 		glog.Errorf("Failed to list minions: %v", err)
 	}
 	for ix, node := range nodes.Items {
-		serversToValidate[fmt.Sprintf("node-%d", ix)] = apiserver.Server{Addr: node.Name, Port: ports.KubeletPort, Path: "/healthz", EnableHTTPS: true}
+		nodeAddr := node.Name
+		serversToValidate[fmt.Sprintf("node-%d", ix)] = apiserver.Server{Addr: nodeAddr, HostHeader: node.Name, Port: ports.KubeletPort, Path: "/healthz", EnableHTTPS: true}
```

Comment:
In #7114 I change nodeAddr to something derived from the NodeAddresses, which is (in general) going to be different from the node.Name.  Probably should have done this as one PR.